docs/wcs/index.rst

```diff
@@ -109,60 +147,95 @@ the WCS cards in its primary header to transform.
+.. toctree::
+   :maxdepth: 1
 
-.. note::
-    The members of the WCS object correspond roughly to the key/value
-    pairs in the FITS header.  However, they are adjusted and
-    normalized in a number of ways that make performing the WCS
-    transformation easier.  Therefore, they can not be relied upon to
-    get the original values in the header.  To build up a FITS header
-    directly and specifically, use `astropy.io.fits.Header` directly.
+   wcsapi.rst
 
-.. _wcslint:
+Legacy Interface
+----------------
 
-Validating the WCS keywords in a FITS file
-------------------------------------------
-
-Astropy includes a commandline tool, ``wcslint`` to check the WCS
-keywords in a FITS file::
-
-    > wcslint invalid.fits
-    HDU 1:
-      WCS key ' ':
-        - RADECSYS= 'ICRS ' / Astrometric system
-          RADECSYS is non-standard, use RADESYSa.
-        - The WCS transformation has more axes (2) than the image it is
-          associated with (0)
-        - 'celfix' made the change 'PV1_5 : Unrecognized coordinate
-          transformation parameter'.
-
-    HDU 2:
-      WCS key ' ':
-        - The WCS transformation has more axes (3) than the image it is
-          associated with (0)
-        - 'celfix' made the change 'In CUNIT2 : Mismatched units type
-          'length': have 'Hz', want 'm''.
-        - 'unitfix' made the change 'Changed units: 'HZ      ' -> 'Hz''.
-
-Bounds checking
----------------
+The following example shows how to load an imaging WCS from an extension HDU.
+The steps can be generalized to the case of a FITS file of any dimensions::
+
+    >>> from astropy.wcs import WCS
+    >>> from astropy.io import fits
+    >>> from astropy.utils.data import get_pkg_data_filename
+    >>> fn = get_pkg_data_filename('data/j94f05bgq_flt.fits', package='astropy.wcs.tests')
+    >>> hdul = fits.open(fn)
+    >>> hdul.info()  #doctest: +SKIP
+    Filename: /root/project/.tox/py36-test/lib/python3.6/site-packages/astropy/wcs/tests/data/j94f05bgq_flt.fits
+    No.    Name      Ver    Type      Cards   Dimensions   Format
+      0  PRIMARY       1 PrimaryHDU     251   ()
+      1  SCI           1 ImageHDU       184   (1, 1)   float32
+      2  ERR           1 ImageHDU        69   ()
+      3  DQ            1 ImageHDU        69   ()
+      4  SCI           2 ImageHDU       184   (1, 1)   float32
+      5  ERR           2 ImageHDU        69   ()
+      6  DQ            2 ImageHDU        69   ()
+    >>> data = hdul[1].data
+    >>> hdr = hdul[1].header
+    >>> wcs = WCS(hdr)
+    >>> wcs  #doctest: +SKIP
+    WCS Keywords
+
+    Number of WCS axes: 2
+    CTYPE : 'RA---TAN-SIP'  'DEC--TAN-SIP'
+    CRVAL : 5.63056810618  -72.05457184278998
+    CRPIX : 2048.0  1024.0
+    CD1_1 CD1_2  : 1.29056256197165e-05  5.95309123310338e-06
+    CD2_1 CD2_2  : 5.0220581265601e-06  -1.2644774105568e-05
+    NAXIS : 1  1
+
+It is a good habit to check how many extensions are there, in case the FITS file is
+of the multi-extension type.
```

Comment:
Ah, this is actually the example I was thinking of putting under `A Simple Example` above (together with the two calls  `pixel_to_world` and `world_to_pixel`. 

And move all other example above here, under  the `Legacy Interface` section . Then move the text in `Legacy Interface` to a separate file and link to it in the `Using astropy.wcs` section.